core/src/main/java/com/exadel/aem/toolkit/core/injectors/BaseInjector.java

```diff
@@ -72,8 +79,13 @@ public final Object getValue(
         if (Objects.isNull(value)) {
             logNullValue(element, annotation);
         }
+        boolean isFallbackValue = value instanceof CastResult && ((CastResult) value).isFallback();
```

Comment:
This logic looks a bit tangled. Consider the following way:
```
Object effectiveValue = value instanceof CastResult ? ((CastResult) value).getValue() : value
boolean isFallback = value instanceof CastResult && ((CastResult) value).isFallback();
if (effectiveValue == null || isFallback) {
    Object defaultValue = getDefaultValue(element.getDeclaredAnnotation(Default.class));
    effectiveValue = CastUtil.toType(defaultValue, type).getValue();
}
return effectiveValue;
```

== less conditions to bear in mind when reading

Reply:
fixed, thank you